dre-service/src/main/java/io/cdap/re/YARERulebookHandler.java

```diff
@@ -437,16 +504,24 @@ public void addRuleToRb(HttpServiceRequest request, HttpServiceResponder respond
   }
 
   @DELETE
-  @Path("rulebooks/{rulebook-id}/rules/{rule-id}")
+  @Path("contexts/{context}/rulebooks/{rulebook-id}/rules/{rule-id}")
   @TransactionPolicy(value = TransactionControl.EXPLICIT)
   public void deleteRuleFromRb(HttpServiceRequest request, HttpServiceResponder responder,
+                               @PathParam("context") String namespace,
                                @PathParam("rulebook-id") String rbId, @PathParam("rule-id") String id) {
     TransactionRunners.run(getContext(), context -> {
       try {
+        NamespacedId rulebookNamespacedId = getNamespacedId(namespace, rbId);
+        NamespacedId ruleNamespacedId = getNamespacedId(namespace, id);
         RulesDB rulesDB = RulesDB.get(context);
-        rulesDB.removeRuleFromRulebook(rbId, id);
-
-        ServiceUtils.success(responder, String.format("Successfully removed rule '%s' to rulebook '%s'", id, rbId));
+        rulesDB.removeRuleFromRulebook(rulebookNamespacedId, ruleNamespacedId);
+
+        ServiceUtils.success(responder, String
+          .format("Successfully removed rule '%s' to rulebook '%s'", ruleNamespacedId.getId(),
+                  rulebookNamespacedId.getId()));
+      } catch (NamespaceNotFoundException nnfe) {
```

Comment:
there is a lot of duplicate code in these endpoints, which makes it difficult to ensure that they all behave correctly. 

It's better to extract the exception handling and namespace lookup into a separate method, similar to https://github.com/data-integrations/wrangler/blob/develop/wrangler-service/src/main/java/io/cdap/wrangler/service/common/AbstractWranglerHandler.java#L174

Reply:
Yes, I looked at that method before. However, each endpoint has different exceptions that they catch and error messages. The only common exception to all endpoints is catching an `Exception` and only some endpoints catch `NamespaceNotFoundException`. Is there a different way to extract common functionality? Or am I looking at the method you suggested incorrectly?

Reply:
in that method, it catches StatusCodeException. That exception is a base class that the various NotFound, AlreadyExists, etc. exceptions extend. Each one corresponds to a different http response code, which the method uses to send the response.